docs/processors/builtin.mdx

```diff
@@ -0,0 +1,234 @@
+
+### filterfieldpayload
+
+Works in the same way as [filterfieldkey](#filterfieldkey), except that it's applied to `Record.Payload.After`.
+
+### hoistfieldkey
+
+`hoistfieldkey` works in the following way:
+
+- If the key is raw, transforms it into structured data by creating a map with the hoisted field and
+  raw data as the value.
+- If the key is structured, wraps it using the specified field name in a map.
+
+#### Configuration
+
+- `field` (Required)
+
+  Name of the field which replaces the entire key.
+
+### hoistfieldpayload
+
+`hoistfieldpayload` builds the same processor as [hoistfieldkey](#hoistfieldkey), except that it operates on the field
+`Record.Payload.After`.
+
+### httprequest
+
+`httprequest` replaces a record's `Payload.After` field with the response of the HTTP request. The response body is
+wrapped in `RawData`.
+
+#### Configuration
+
+- `url` (Required)
+
+  URL used in the HTTP request.
+- `method` (Optional)
+
+  HTTP request method to be used. Defaults to `POST`.
+- `backoffRetry.count` (Optional, integer)
+
+  Maximum number of retries when backing off following an error. Defaults to 0.
+- `backoffRetry.factor` (Optional, integer)
+
+  The multiplying factor for each increment step. Defaults to 2.
+- `backoffRetry.min` (Optional, [duration string](https://pkg.go.dev/time#ParseDuration))
+
+  Minimum waiting time before retrying. Defaults to `100ms`.
+- `backoffRetry.max` (Optional, [duration string](https://pkg.go.dev/time#ParseDuration))
+  
+  Maximum waiting time before retrying. Defaults to `5s`.
+
+### insertfieldkey
+
+Inserts a field into a record's key. It's possible to insert a static value or the record's position. If the field
+already
+exists, it's overwitten. Raw keys are not supported yet.
+
+#### Configuration
+
+- `static.field` (Required, if `position.field` is not set)
+
+  Name of the field into which the value in `static.value` will be inserted.
+- `static.value` (Required, if `static.field` is set)
+
+  Value to be inserted.
+- `position.field` (Required, if `static.field` is not set)
+
+  Name of the field into which the record's position will be inserted.
+
+### insertfieldpayload
+
+Works in the same way as [insertfieldkey](#insertfieldkey), except that it's applied to `Record.Payload.After`.
+
+### maskfieldkey
+
+`maskfieldkey` replaces a field in a key with structured data with the zero value of its type. If the field is a string,
+an interger, or a float it's possible to define a replacement value. Raw keys are not supported yet.
+
+#### Configuration
+
+- `field` (Required)
+
+  Name of the field to be masked.
+
+- `replacement` (Optional)
+
+  Replacement value. Defaults to the field type's zero value (e.g. empty string for strings, 0 for integers etc.)
+
+### maskfieldpayload
+
+Works in the same way as [maskfieldpayload](#maskfieldpayload), except that it's applied to `Record.Payload.After`.
+
+### parsejsonkey
+
+Transforms the record's key into structured data, i.e. a JSON object.
```

Comment:
I think this processor and `parsejsonpayload` need more details, maybe something like:
```suggestion
Transforms the record's raw data key (has to be JSON formatted) into structured data, i.e. a JSON object.
```

Reply:
@hariso looks like you didn't notice the comment, we need to update the payload processor too 👀

Reply:
@maha-hajja Thinking again about this... IIUC, the `parsejson` pair of processors will parse the JSON even when the data is structured, which means that we don't need to say "transform the record's **raw data** key/payload.after...". Is my understanding correct? I agree that the note about that it has to be JSON formatted should be there.

Reply:
@hariso if the data is already structured then it's just returned as is